src/prototree/models.py

```diff
@@ -160,21 +160,27 @@ def predict(self, x: torch.Tensor) -> torch.Tensor:
         return torch.argmax(self.predict_probs(x), dim=-1)
 
 
+@dataclass
+class NodeSimilarity:
+    similarity: ImageProtoSimilarity
+    node: InternalNode
+
+
 @dataclass
 class LeafRationalization:
-    ancestor_similarities: list[ImageProtoSimilarity]
+    ancestor_sims: list[ImageProtoSimilarity]
```

Comment:
Is there any way to validate whether the ancestor sims passed actually contain ancestors? If yes, one could use post-init or (even better, I think) pydantic validators

Reply:
There was a misunderstanding here. I didn't mean the basic check that the list is not empty (although that's also useful, I guess) but the fact the type here is a general ImageProtoSimilarity, however the implementation assumes that similarities of only the ancestors of the leaf are passed, not just any similarities. Things would go wrong in non-obvious ways if the user makes a mistake and passes a wrong list here. I guess it wouldn't be too hard to make this mistake, right? That's why I wondered if this can be validated in some way. Alternatively, if it's difficult to validate, one could maybe turn this into a private class or add a docstring discouraging the user from using it directly, if they don't know what they're doing